File: frontend1/src/types/TourneyMatch.ts
import { Team } from './Team';
import { MatchGame } from './MatchGame';
import { Tournament } from './Tournament';

export type TourneyMatch = {
  matchId: number;
  tourneyId: number;
  lanes: string;
  oddLaneTeamId: number;
  evenLaneTeamId: number;
  evenLaneTeam: Team;
  matchGames: MatchGame[];
  oddLaneTeam: Team;
  tourney: Tournament;
};

File: backend/BowlerAPI/Data/TourneyMatch.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BowlerAPI.Data;

public partial class TourneyMatch
{
    [Key]
    [Required]
    public int MatchId { get; set; }
    [ForeignKey("Tournament")]
    public int? TourneyId { get; set; }

    public string? Lanes { get; set; }
    [ForeignKey("Team")]
    public int? OddLaneTeamId { get; set; }
    [ForeignKey("Team")]
    public int? EvenLaneTeamId { get; set; }

    public virtual Team? EvenLaneTeam { get; set; }

    public virtual ICollection<MatchGame> MatchGames { get; set; } = new List<MatchGame>();

    public virtual Team? OddLaneTeam { get; set; }

    public virtual Tournament? Tourney { get; set; }
}
